addons/mrp/models/mrp_production.py

```diff
@@ -301,14 +301,44 @@ def _set_date_planned_finished(self):
         if self.date_planned_finished:
             self.move_finished_ids.write({'date_expected': self.date_planned_finished})
 
+    def _search_planned(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise UserError(_('Operation not supported'))
+        value = value if operator == '=' else not value
+        self.env.cr.execute('''
+              SELECT wo.production_id
+                FROM mrp_workorder wo
+               {}
+            GROUP BY wo.production_id
+              HAVING bool_and(wo.date_planned_start IS NOT NULL AND wo.date_planned_finished IS NOT NULL)
+        '''.format("WHERE wo.production_id IN %(ids)s" if self else ""),
+            {'ids': tuple(self.ids)}
+        )
+        return [('id', 'in' if value else 'not in', [r[0] for r in self._cr.fetchall()])]
+
+    def _search_partially_planned(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise UserError(_('Operation not supported'))
+        value = value if operator == '=' else not value
+        self.env.cr.execute('''
+              SELECT wo.production_id
+                FROM mrp_workorder wo
+                JOIN mrp_production prod ON wo.production_id = prod.id
+               WHERE prod.state != 'draft'
+                 {}
+            GROUP BY wo.production_id
+              HAVING bool_or(wo.date_planned_start IS NOT NULL AND wo.date_planned_finished IS NOT NULL)
+        '''.format("AND wo.production_id IN %(ids)s" if self else ""),
+            {'ids': tuple(self.ids)}
+        )
+        return [('id', 'in' if value else 'not in', [r[0] for r in self._cr.fetchall()])]
+
     def _compute_is_planned(self):
+        planned_ids = self.search([('is_planned', '=', True)])
+        partially_planned_ids = self.search([('is_partially_planned', '=', True)])
         for production in self:
-            if production.workorder_ids:
-                production.is_planned = all(wo.date_planned_start and wo.date_planned_finished for wo in production.workorder_ids)
-                production.is_partially_planned = any(wo.date_planned_start and wo.date_planned_finished for wo in production.workorder_ids if production.state != 'draft')
-            else:
-                production.is_planned = False
-                production.is_partially_planned = False
+            production.is_planned = production in planned_ids
+            production.is_partially_planned = production in partially_planned_ids
```

Comment:
I guess I could leave the method `_compute_is_planned` untouched, but since I have to implement `_search_planned` and `_search_partially_planned` because of a filter, I thought I might change it here as well.